### PR TITLE
Update DDIO and related to it's code OSIRIS

### PR DESCRIPTION
**Pull request by winterheart**
## Pull Request Type

- [ ] GitHub Workflow changes
- [ ] Documentation or Wiki changes
- [ ] Build and Dependency changes
- [x] Runtime changes
  - [ ] Render changes
  - [ ] Audio changes
  - [ ] Input changes
  - [ ] Network changes
  - [x] Other changes

### Description

Remove unused functions ddio_GetParentPath() and ddio_CleanPath(). Implement new crossplatform ddio_GetTmpFileName() to replace ddio_GetTempFileName().

Optimize module loading in OSIRIRS: use std::filesystem::path on most parts, change OSIRIS_Extracted_scripts to std::map that removes some fixed size restrictions and optimize search of extracted modules.

### Checklist

- [x] I have tested my changes locally and verified that they work as intended.
- [x] I have documented any new or modified functionality.
- [x] I have reviewed the changes to ensure they do not introduce any unnecessary complexity or duplicate code.
- [x] I understand that by submitting this pull request, I am agreeing to license my contributions under the project's license.

### CODE REVIEW

**Comment by winterheart** [comment]
Benchmark of ddio_GetTmpFileName() vs ddio_GetTempFileName() on Linux (Release build):

```
./ddio_bench --benchmark_repetitions=10
2024-09-17T13:35:02+03:00
Running ./ddio_bench
Run on (12 X 4074.82 MHz CPU s)
CPU Caches:
  L1 Data 32 KiB (x6)
  L1 Instruction 32 KiB (x6)
  L2 Unified 256 KiB (x6)
  L3 Unified 12288 KiB (x1)
Load Average: 0.34, 0.46, 0.88
--------------------------------------------------------------
Benchmark                    Time             CPU   Iterations
--------------------------------------------------------------
ddio_TempOld              2688 ns         2545 ns       271999
ddio_TempOld              2693 ns         2561 ns       271999
ddio_TempOld              2714 ns         2575 ns       271999
ddio_TempOld              3853 ns         3705 ns       271999
ddio_TempOld              5084 ns         4936 ns       271999
ddio_TempOld              5103 ns         4950 ns       271999
ddio_TempOld              5279 ns         5125 ns       271999
ddio_TempOld              5389 ns         5237 ns       271999
ddio_TempOld              5408 ns         5256 ns       271999
ddio_TempOld              5428 ns         5279 ns       271999
ddio_TempOld_mean         4364 ns         4217 ns           10
ddio_TempOld_median       5093 ns         4943 ns           10
ddio_TempOld_stddev       1236 ns         1230 ns           10
ddio_TempOld_cv          28.33 %         29.17 %            10
ddio_TempNew              2071 ns         2070 ns       338686
ddio_TempNew              2068 ns         2066 ns       338686
ddio_TempNew              2067 ns         2066 ns       338686
ddio_TempNew              2069 ns         2068 ns       338686
ddio_TempNew              2068 ns         2067 ns       338686
ddio_TempNew              2064 ns         2063 ns       338686
ddio_TempNew              2068 ns         2067 ns       338686
ddio_TempNew              2063 ns         2062 ns       338686
ddio_TempNew              2071 ns         2070 ns       338686
ddio_TempNew              2070 ns         2069 ns       338686
ddio_TempNew_mean         2068 ns         2067 ns           10
ddio_TempNew_median       2068 ns         2067 ns           10
ddio_TempNew_stddev       2.80 ns         2.75 ns           10
ddio_TempNew_cv           0.14 %          0.13 %            10
```

File: ddio/file.cpp [hunk 1]
```diff
@@ -156,3 +159,36 @@ void ddio_DoForeachFile(const std::filesystem::path &search_path, const std::reg
...
+  const char *ext = ".tmp";
+  std::filesystem::path result;
+  size_t len_result = strlen((basedir / prefix).u8string().c_str());
+  char *random_name = (char *)mem_malloc(len_result + len + strlen(ext) + 1);
+  strncpy(random_name, (basedir / prefix).u8string().c_str(), len_result);
+
+  srand(D3::ChronoTimer::GetTimeMS());
+
+  int tries = 20;
+  while (tries > 0) {
```
Comment (Lgt2x) [inline ddio/file.cpp:179]: I'm honestly not fond of this approach of retrying until we get a correct name (even if it's unlikely we have collisions). A better way to do it overall would be to use a hash (md5) of the file we mirror if we can; in my understanding these temp files correspond to unique uncompressed graphics, sounds resources, etc. What do you think of that?

**Review by Lgt2x — COMMENTED** [review]
nice job, ddio is starting to look good! `ddio_ff_SetCoopLevel` can also be removed apparently

File: ddio/file.cpp (hunk 1 above)
Comment (winterheart) [reply ddio/file.cpp:179]: Well, collisions can occurs, especially with current Linux `ddio_GetTempFileName()`, if generate names too quickly. Luckily we don't have to do in real play, and there no such issues.

Hash based names has limited use in project. Literally only shared library extraction code can use this feature.

Comment (Lgt2x) [reply ddio/file.cpp:179]: ok I got it, all fine then!

**Review by Lgt2x — APPROVED** [review]
No problem in game, code is clean

**Pull request merged**